### DISCUSSION TITLE
[schema] Circular dependencies

### DISCUSSION THREAD

**Discussion by mdartic** [feathersjs/feathers#3018] [Q&A]
Following the tip in https://github.com/feathersjs/feathers/issues/2993, 

I'm blocked when I want to specify schemas with circular dependencies.

Example :
I have users, and workspaces.
Each `user` can have several `workspace`, but one `workspace` belongs to one `user`.

So I would like to specify my schemas like this : 

```ts
// user.schema.ts
import { workspaceSchema } from "../workspace/workspace.schema"
const userSchema = Type.Object({
  // ...
  workspace: Type.Optional(Type.Ref(workspaceSchema))
}, { $id: 'User', additionalProperties: false })

// workspace.schema.ts
import { userSchema } from "../users/user.schema"

const workspaceSchema = Type.Object({
  // ...
  user: Type.OptionalType.Ref(userSchema))
}, { $id: 'Workspace', additionalProperties: false })
```

I find an issue on typebox talking about this https://github.com/sinclairzx81/typebox/issues/106

But I can't figure out to make this work in Feathers...

Do you have any idea ?

Actually, I'm faking the result, so it's a `workspace: Type.Optional(Type.Any())`.

For two models, that could work, but for nested relations, this seems a nightmare :-)

**Comment by daffl** [comment]
There is no additional abstraction between TypeBox and Feathers so what is described in the issue should work. I usually try to avoid circular dependencies because they always cause more issues than they help solve and instead try to design a unidirectional data model.
